core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -85,6 +85,9 @@ private[spark] class Executor(
     env.blockManager.initialize(conf.getAppId)
   }
 
+  private val executorMetrics: ExecutorMetrics = new ExecutorMetrics
+  executorMetrics.setHostname(Utils.localHostName)
```

Comment:
Sorry that I forgot the reason why I didn't get the port from `rpcEnv` originally, but it seems we can get the port from it directly. Let me try to add it back.